node/src/command.rs

```diff
@@ -52,20 +51,29 @@ impl SubstrateCli for Cli {
     }
 
     fn load_spec(&self, id: &str) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
-        Ok(match id {
-            "dev" => Box::new(chain_spec::development_config()),
-            "local" => Box::new(chain_spec::local_testnet_config()),
-            "" | "main" => Box::new(chain_spec::main_config()),
-            "arcadia" => Box::new(chain_spec::arcadia_config()),
-            "staking-local" => Box::new(chain_spec::local_staking_config()),
-            path => Box::new(chain_spec::ChainSpec::from_json_file(
-                std::path::PathBuf::from(path),
-            )?),
-        })
+        if self.run.runtime.is_staking_runtime() {
+            Ok(match id {
+                "dev" => Box::new(chain_spec::cs_staking::development_config()),
+                "" | "local" => Box::new(chain_spec::cs_staking::local_staking_config()),
+                path => Box::new(chain_spec::cs_staking::ChainSpec::from_json_file(
+                    std::path::PathBuf::from(path),
+                )?),
+            })
+        } else {
+            Ok(match id {
+                "dev" => Box::new(chain_spec::cs_main::development_config()),
+                "local" => Box::new(chain_spec::cs_main::local_testnet_config()),
+                "" | "main" => Box::new(chain_spec::cs_main::main_config()),
+                "arcadia" => Box::new(chain_spec::cs_main::arcadia_config()),
+                path => Box::new(chain_spec::cs_staking::ChainSpec::from_json_file(
+                    std::path::PathBuf::from(path),
+                )?),
+            })
+        }
     }
 
     fn native_runtime_version(_: &Box<dyn ChainSpec>) -> &'static RuntimeVersion {
-        &runtime_main::VERSION
+        &main_runtime::VERSION
     }
```

Comment:
depends on the runtime right>

Reply:
Good one, restored back.
Indeed its my bad, while implementing below chain identification api,
I was using `starts_with()`, so changed the name of crate.

```
fn is_runtime_staking(&self) -> bool {
	self.id().to_lowercase().starts_with("staking")
}
```

But finally id is picked from chain spec definition ...

```
ChainSpec::from_genesis(
	"Staking Local Testnet",
	"staking_local_testnet",
	ChainType::Local,
	local_staking_genesis,
	vec![],
	None,
	Some("nodl"),
	Some(build_local_properties()),
	Default::default(),
)
```